proto/opamp.proto

```diff
@@ -264,6 +264,20 @@ message OpAMPConnectionSettings {
     // This field is optional: if omitted the client SHOULD NOT use a client-side certificate.
     // This field can be used to perform a client certificate revocation/rotation.
     TLSCertificate certificate = 3;
+
+    // The Agent MUST periodically send an AgentToServer message if this field is set. At a minimum
+    // the instance_uid field MUST be set. It is recommended that the Agent also set ComponentHealth
+    // as well.
+    //
+    // A Polling-based HTTP Client MUST use the value as polling interval.
+    //
+    // A heartbeat is used to keep a load balancer connection active and inform the server that the Agent
+    // is still alive and active.
+    //
+    // This field is optional:
+    // if omitted, a default heartbeat interval of 30 seconds should be used.
+    // if set to zero, the heartbeat ability will be disabled.
+    optional uint64 heartbeat_interval_seconds = 4;
```

Comment:
IIRC, some Protobuf implementations can't differentiate between "not present" and zero values (e.g. Gogoproto). Can we have a different way that does not rely on this capability?

Reply:
yes, i can remove that bit now that I added heartbeat as part of the AgentCapabilities Enum

Reply:
from past discussions we had decided that we were going to use the capability to determine whether or not heartbeats should be disabled. I don't think we should have multiple ways of disabling this.

Reply:
The operation is not entire clear to me. Who is making a decision to disable the heartbeats? Is it the Server or the Client?

Reply:
ah i see what you're saying. I think the flow is this:
1. client reports that it can send heartbeats
2. server sets a non-zero heartbeat interval that the client respects

With this in mind, either the client or the server can disable heartbeats at any time.

Reply:
OK. I think we need to describe the flows and possible options (with heartbeats, without heartbeats, client deciding, sever deciding, etc), what's the initial state, etc. 

It may be even worth showing sequence diagrams if plain text is not clear enough.